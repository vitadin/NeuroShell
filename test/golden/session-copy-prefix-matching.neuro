# Test session copy with prefix matching functionality
\session-new[system="You are a project management assistant"] project_management_main
\session-add-usermsg How should I organize my project tasks?
\session-add-assistantmsg I recommend using a structured approach...

# Test prefix matching by name (should match project_management_main)
\session-copy[source_session_name=project_management, target_session_name=pm_backup]
\echo Copied from: ${#source_session_name}
\echo Copied to: ${#session_name}
\echo Message count: ${#message_count}

# Test with even shorter prefix
\session-copy[source_session_name=project, target_session_name=another_copy]
\echo Second copy from: ${#source_session_name}
\echo Second copy to: ${#session_name}