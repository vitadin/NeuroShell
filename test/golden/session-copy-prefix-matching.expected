# Test session copy with prefix matching functionality
Created session 'project_management_main' (ID: 00000001)
Added user message to session '00000001-0000-4000-8000-000000000001'
Added assistant message to session '00000001-0000-4000-8000-000000000001'
# Test prefix matching by name (should match project_management_main)
Copied session 'project_management_main' to 'pm_backup' (ID: 00000004) with 2 messages
Copied from: project_management_main
Copied to: pm_backup
Message count: 2
# Test with even shorter prefix
Copied session 'project_management_main' to 'another_copy' (ID: 00000007) with 2 messages
Second copy from: project_management_main
Second copy to: another_copy